src/app/ReadClient.cpp

```diff
@@ -70,6 +71,26 @@ ReadClient::~ReadClient()
 
 void ReadClient::Close(CHIP_ERROR aError)
 {
+    if (aError != CHIP_NO_ERROR)
+    {
+        if (mReadPrepareParams.mShouldResubscribe)
+        {
+            CHIP_ERROR err = Resubscribe();
+            if (mReadPrepareParams.mShouldResubscribe && err == CHIP_NO_ERROR)
+            {
+                mIsInitialReport           = true;
```

Comment:
Please commonize this entire bit of logic and re-use it in the constructor too.

Reply:
Updated, thanks